app/util/job.ts

```diff
@@ -38,11 +38,9 @@ export default async function cancelAndSaveJob(
         job.validateStatus();
         job.cancel(message);
         await job.save(tx);
-        const updatedAt = new Date();
-        await tx('work_items')
-          .where({ jobID: job.jobID })
-          .whereIn('status', [WorkItemStatus.READY, WorkItemStatus.RUNNING])
-          .update({ status: WorkItemStatus.CANCELED, updatedAt });
+        await updateWorkItemsStatusByJobId(
+          tx, job.jobID, [WorkItemStatus.READY, WorkItemStatus.RUNNING], WorkItemStatus.CANCELED,
```

Comment:
I know this isn't quite your change, but if something is `RUNNING` and then gets `CANCELED` like this, presumably whatever call was working on it will call back and try to update its status.  Is that behavior handled properly?

Reply:
This is what happens in that case:

```
// If the job was already canceled then send 400 response
    if (job.status === JobStatus.CANCELED) {
      res.status(409).send('Job was already canceled.');
```

Reply:
Actually, the status of the work item would get updated again (undesirable) on the call back. I think I can just move the update invocation around to account for this.